spec/config_manager/http_project_config_manager_spec.rb

```diff
@@ -17,141 +17,166 @@
-        url: 'https://cdn.optimizely.com/datafiles/QBw9gFM8oTn7ogY9ANCC1z.json'
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        url: 'https://cdn.optimizely.com/datafiles/valid_sdk_key.json'
       )
 
-      until http_project_config_manager.ready?; end
-      expect(http_project_config_manager.config).not_to eq(nil)
+      until @http_project_config_manager.ready?; end
+      expect(@http_project_config_manager.config).to be_a Optimizely::ProjectConfig
     end
 
     it 'should get project config when sdk_key is given' do
-      http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
-        sdk_key: 'QBw9gFM8oTn7ogY9ANCC1z'
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: 'valid_sdk_key'
       )
 
-      until http_project_config_manager.ready?; end
-      expect(http_project_config_manager.config).not_to eq(nil)
+      until @http_project_config_manager.ready?; end
+      expect(@http_project_config_manager.config).to be_a Optimizely::ProjectConfig
     end
 
     it 'should get project config when sdk_key and valid url_template is given' do
-      http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
-        sdk_key: 'QBw9gFM8oTn7ogY9ANCC1z',
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: 'valid_sdk_key',
         url_template: 'https://cdn.optimizely.com/datafiles/%s.json'
       )
-      until http_project_config_manager.ready?; end
+      until @http_project_config_manager.ready?; end
 
-      expect(http_project_config_manager.config).not_to eq(nil)
+      expect(@http_project_config_manager.config).to be_a Optimizely::ProjectConfig
     end
 
     it 'should get instance ready immediately when datafile is provided' do
-      start = Time.now
-      http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
-        sdk_key: 'QBw9gFM8oTn7ogY9ANCC1z',
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: 'valid_sdk_key',
         datafile: config_body_JSON,
         start_by_default: false
       )
-      until http_project_config_manager.ready?; end
-      finish = Time.now
-      expect(finish - start).to be < 1
+      # This expectation on main thread confirms that instance got ready immediately
+      expect(@http_project_config_manager.ready?).to be true
     end
 
     it 'should wait to get project config when datafile is not provided' do
-      http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
-        sdk_key: 'QBw9gFM8oTn7ogY9ANCC1z',
+      @http_project_config_manager = Optimizely::HTTPProjectConfigManager.new(
+        sdk_key: 'valid_sdk_key',
         start_by_default: false
       )
       start = Time.now
-      http_project_config_manager.start!
-      until http_project_config_manager.ready?; end
+      @http_project_config_manager.start!
```

Comment:
can we remove timing difference logic and see the mocked method shouldn't get called. It's oK to me

Reply:
Found a nicer way to test blocking timeout part. Revised test.